dadapy/clustering.py

```diff
@@ -250,7 +251,9 @@ def compute_clustering_pure_python(  # noqa: C901
             print("Clustering started")
 
         # Make all values of log_den positives (this is important to help convergence)
-        log_den_min = np.min(self.log_den)
+        # even when substracting the value Z*log_den_err
+        log_den_min = np.min(self.log_den - Z * self.log_den_err)
```

Comment:
@imacocco probably the problem you described to be this morning is fixed by this line

Reply:
Great, I'll check how it works as soon as the PR is accepted